src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Performance/DoNotGuardDictionaryRemoveByContainsKeyTests.cs

```diff
@@ -0,0 +1,298 @@
+
+using System.Threading.Tasks;
+using Xunit;
+
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Performance.DoNotGuardDictionaryRemoveByContainsKey,
+    Microsoft.NetCore.Analyzers.Performance.DoNotGuardDictionaryRemoveByContainsKeyFixer>;
+using VerifyVB = Test.Utilities.VisualBasicCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Performance.DoNotGuardDictionaryRemoveByContainsKey,
+    Microsoft.NetCore.Analyzers.Performance.DoNotGuardDictionaryRemoveByContainsKeyFixer>;
+
+namespace Microsoft.NetCore.Analyzers.Performance.UnitTests
+{
+    public class DoNotGuardDictionaryRemoveByContainsKeyTests
+    {
+        #region Reports Diagnostic
+        [Fact]
+        public async Task RemoveIsTheOnlyStatement_ReportsDiagnostic_CS()
+        {
+            string source = @"
+" + CSUsings + @"
+namespace Testopolis
+{
+    public class MyClass
+    {
+        private readonly Dictionary<string, string> MyDictionary = new Dictionary<string, string>();
+
+        public MyClass()
+        {
+            if ([|MyDictionary.ContainsKey(""Key"")|])
+                MyDictionary.Remove(""Key"");
+        }
+    }
+}";
+
+            string fixedSource = @"
+" + CSUsings + @"
+namespace Testopolis
+{
+    public class MyClass
+    {
+        private readonly Dictionary<string, string> MyDictionary = new Dictionary<string, string>();
+
+        public MyClass()
+        {
+            MyDictionary.Remove(""Key"");
+        }
+    }
+}";
+            await VerifyCS.VerifyCodeFixAsync(source, fixedSource);
+        }
+
+        [Fact]
+        public async Task RemoveIsTheOnlyStatementInABlock_ReportsDiagnostic_CS()
+        {
+            string source = @"
+" + CSUsings + @"
+namespace Testopolis
+{
+    public class MyClass
+    {
+        private readonly Dictionary<string, string> MyDictionary = new Dictionary<string, string>();
+
+        public MyClass()
+        {
+            if ([|MyDictionary.ContainsKey(""Key"")|])
+            {
+                MyDictionary.Remove(""Key"");
+            }
+        }
+    }
+}";
+
+            string fixedSource = @"
+" + CSUsings + @"
+namespace Testopolis
+{
+    public class MyClass
+    {
+        private readonly Dictionary<string, string> MyDictionary = new Dictionary<string, string>();
+
+        public MyClass()
+        {
+            MyDictionary.Remove(""Key"");
+        }
+    }
+}";
+
+            await VerifyCS.VerifyCodeFixAsync(source, fixedSource);
+        }
+
+        [Fact]
+        public async Task HasElseBlock_NoDiagnostic_CS()
+        {
+            string source = @"
+" + CSUsings + @"
+namespace Testopolis
+{
+    public class MyClass
+    {
+        private readonly Dictionary<string, string> MyDictionary = new Dictionary<string, string>();
+
+        public MyClass()
+        {
+            if (MyDictionary.ContainsKey(""Key""))
+            {
+                MyDictionary.Remove(""Key"");
```

Comment:
Is it possible to fix this one as it has only one statement in the `true` section? As everything in the else block can be kept as is:

```cs
    public MyClass()
    {
        if (!MyDictionary.Remove("Key"))
        {
           throw new Exception("Key doesn't exist");
        }
    }
```
But it is optional, we can add this to the future enhancement

Reply:
Or i think we should at least warn in this case, even not suggesting a fixer, what you think?

Reply:
I'm guessing you mean the test `NegatedCondition_NoDiagnostic_CS`? If so, yup, reporting a diagnostic on that now.

Reply:
Oh, I see you mean the Else test. I’ll give it a shot.

Reply:
Also done.